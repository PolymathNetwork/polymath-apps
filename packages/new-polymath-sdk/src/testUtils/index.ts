import Web3PromiEvent from 'web3-core-promievent';
import { GenericContract } from '~/LowLevel/types';
import { PostTransactionResolver } from '~/PostTransactionResolver';

export class MockedContract<T extends GenericContract> {
  public autoResolve: boolean;
  public errorMsg?: string;
  public fakeTxOnePromiEvent = new Web3PromiEvent();
  public fakeTxTwoPromiEvent = new Web3PromiEvent();
  public failureTxPromiEvent = new Web3PromiEvent();

  public fakeTxOne = jest.fn(async () => {
    return () => {
      if (this.autoResolve) {
        this.fakeTxOnePromiEvent.resolve();
      }
      return this.fakeTxOnePromiEvent.eventEmitter;
    };
  });
  public fakeTxTwo = jest.fn(async () => {
    return () => {
      if (this.autoResolve) {
        this.fakeTxTwoPromiEvent.resolve();
      }
      return this.fakeTxTwoPromiEvent.eventEmitter;
    };
  });

  public failureTx = jest.fn(async () => {
    return () => {
      const err = this.errorMsg || 'Test error';
      this.failureTxPromiEvent.reject(new Error(err));
      return this.failureTxPromiEvent.eventEmitter;
    };
  });

  constructor({
    autoResolve = true,
    errorMsg,
  }: { autoResolve?: boolean; errorMsg?: string } = {}) {
    this.autoResolve = autoResolve;
    this.errorMsg = errorMsg;
  }
}

export const getMockTransactionSpec = (
  method: (args: any) => Promise<any>,
  args: any,
  resolver = async () => {}
) => ({
  method,
  args,
  postTransactionResolver: new PostTransactionResolver(resolver),
});
