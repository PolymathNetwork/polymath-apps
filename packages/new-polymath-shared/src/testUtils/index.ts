export * from './factories';

const originalWindow = {
  ...window,
};

interface WindowWithEthereum extends Window {
  ethereum: {
    networkVersion: string;
    _metamask?: {
      isApproved: () => Promise<boolean>;
    };
    enable(): Promise<any>;
  };
}

interface WindowWithWeb3 extends Window {
  web3: {
    version: string;
  };
}

interface MockEthereumBrowserArgs {
  support?: 'legacy' | 'modern';
  options?: {
    networkId: number;
    loaded: boolean;
  };
}

export function mockEthereumBrowser({
  support = 'modern',
  options = {
    networkId: 1,
    loaded: true,
  },
}: MockEthereumBrowserArgs) {
  const { networkId, loaded } = options;

  if (support === 'modern') {
    window = (Object.assign(window, {
      ethereum: {
        networkVersion: loaded ? `${networkId}` : undefined,
      },
    }) as any) as WindowWithEthereum;
  }
  if (support === 'legacy') {
    window = Object.assign(window, {
      web3: {
        version: networkId,
      },
    } as any) as WindowWithWeb3;
  }

  return {
    restore: () => {
      window = originalWindow;
    },
    load: () => {
      if (support === 'modern') {
        (window as WindowWithEthereum).ethereum.networkVersion = `${networkId}`;
      }
      if (support === 'legacy') {
        (window as WindowWithWeb3).web3.version = `${networkId}`;
      }
    },
  };
}
