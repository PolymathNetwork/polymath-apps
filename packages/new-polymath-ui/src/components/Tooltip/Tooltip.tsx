import React, { Fragment } from 'react';
import { findDOMNode } from 'react-dom';
import PopperJS from 'popper.js';
import TooltipJS, { Options } from 'tooltip.js';
import { styled } from '~/styles';
import * as sc from './styles';

export interface TooltipProps extends Options {
  role?: string;
  placement?: PopperJS.Placement;
  className?: string;
}

export class TooltipComponent extends React.Component<TooltipProps> {
  public static defaultProps = {
    role: 'group',
    placement: 'top',
  };

  private popper?: TooltipJS;

  public getOptions = (): Options => {
    const { placement, role, children, ...otherProps } = this.props;
    const popover = this.getPopover() as HTMLElement;

    return {
      placement,
      title: popover,
      html: true,
      container: document.body,
      popperOptions: {
        modifiers: {
          setState: {
            enabled: true,
            order: 800,
            fn: this.modifier,
          },
        },
      },
      ...otherProps,
    };
  };

  public initPopper = () => {
    if (!this.popper) {
      const popover = this.getPopover();
      const parentNode = popover.parentNode as HTMLElement;

      if (parentNode && getComputedStyle(parentNode).display !== 'none') {
        this.popper = new TooltipJS(parentNode, this.getOptions());
      }
    }
  };

  public destroyPopper = () => {
    if (this.popper) {
      this.popper.dispose();
      this.popper = undefined;
    }
  };

  public componentDidMount() {
    this.initPopper();
  }

  public componentWillUnmount() {
    this.destroyPopper();
  }

  public componentDidUpdate(
    prevProps: TooltipProps & Readonly<{ children?: React.ReactNode }>
  ) {
    if (
      prevProps.placement !== this.props.placement ||
      prevProps.children !== this.props.children
    ) {
      this.destroyPopper();
      this.initPopper();
    }
  }

  public render() {
    const { role, children, className } = this.props;
    return (
      <Fragment>
        <sc.GlobalStyles />
        <div className={className} role={role}>
          {children}
        </div>
      </Fragment>
    );
  }

  private getPopover = () => findDOMNode(this) as NodeSelector & Node & Element;

  private modifier = (data: PopperJS.Data) => {
    data.arrowStyles.color = getComputedStyle(
      this.getPopover()
    ).backgroundColor;
    return data;
  };
}

export const Tooltip = styled(TooltipComponent)<TooltipProps>`
  user-select: auto;
  cursor: auto;
  display: none;

  .tooltip & {
    display: block;
  }
`;
