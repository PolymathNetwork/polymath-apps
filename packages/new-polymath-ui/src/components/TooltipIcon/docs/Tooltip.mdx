---
name: Tooltip
menu: Components
---

import { Playground, PropsTable } from 'docz';

import { Tooltip } from '../';

# Tooltip

<PropsTable of={Tooltip} />

## Basic usage

<Playground>
  <Tooltip triggerText="Hover the icon!">This is a Tooltip.</Tooltip>
</Playground>
