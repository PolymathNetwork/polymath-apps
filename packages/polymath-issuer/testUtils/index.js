import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import * as factories from './factories';

const middlewares = [thunk];
export const mockStore = configureMockStore(middlewares);

export { factories };
