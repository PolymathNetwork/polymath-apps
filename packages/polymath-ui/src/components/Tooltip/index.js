import React, { Fragment } from 'react';
import styled from 'styled-components';
import ReactTooltip from 'react-tooltip';

import Icon from '../Icon';

import InfoIcon from '../../images/icons/Info';

const StyledTooltip = styled(ReactTooltip)`
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.1);
  max-width: 15rem;
  background-color: white;
  padding: 1rem;
  border: 1px solid #dfe3e6;
  border-radius: 0.25rem;
  word-wrap: break-word;
  pointer-events: auto;

  &.place-top:before {
    border-top: 8px solid rgba(0, 0, 0, 0.05);
  }

  &.place-bottom:before {
    border-bottom: 8px solid #dfe3e6;
  }
`;

const StyledText = styled.span`
  vertical-align: middle;
  margin-right: 4px;
`;

const StyledIcon = styled(Icon)`
  color: ${({ theme }) => theme.colors.blue[1]};
`;

const Tooltip = ({ triggerText, children, id }) => (
  <Fragment>
    <StyledText>{triggerText}</StyledText>
    <StyledIcon data-tip data-for={id} Icon={InfoIcon} width="16" height="16" />
    <StyledTooltip
      id={id}
      delayHide={200}
      effect="solid"
      type="light"
      place="bottom"
    >
      {children}
    </StyledTooltip>
  </Fragment>
);

export default Tooltip;

Tooltip.defaultProps = {};
