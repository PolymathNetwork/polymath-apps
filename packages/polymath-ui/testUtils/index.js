import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import theme from '../src/theme';

const middlewares = [thunk];

const customRender = (node, ...opts) => {
  const { rerender, ...result } = render(
    <ThemeProvider theme={theme}>{node}</ThemeProvider>,
    ...opts
  );

  const customRerender = (rerenderNode, ...rerenderOpts) => {
    return rerender(
      <ThemeProvider theme={theme}>{rerenderNode}</ThemeProvider>,
      ...rerenderOpts
    );
  };

  return { rerender: customRerender, ...result };
};

export const mockStore = configureMockStore(middlewares);
export * from 'react-testing-library';
export { customRender as render };
