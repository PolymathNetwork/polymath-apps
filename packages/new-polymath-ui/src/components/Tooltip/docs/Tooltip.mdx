---
name: Tooltip
menu: Components
---

import { Playground, PropsTable } from 'docz';

import { Box } from '~/components/Box';
import { Button } from '~/components/Button';
import { Paragraph } from '~/components/Paragraph';
import { TooltipPrimary } from '~/components/TooltipPrimary';
import { TooltipIcon } from '~/components/TooltipIcon';
import { Tooltip } from '../';

# Tooltip

<PropsTable of={Tooltip} />

## Primitive component

<Playground>
  <Box height={100}>
    <Button>
      Hover me
      <Tooltip>This is a Tooltip.</Tooltip>
    </Button>
  </Box>
</Playground>

## TooltipPrimary

<Playground>
  <Box height={100}>
    <Button>
      Hover me
      <TooltipPrimary>This is a TooltipPrimary.</TooltipPrimary>
    </Button>
  </Box>
</Playground>

## TooltipIcon

<Playground>
  <Box height={100}>
    <Paragraph>
      <span>Label of an input</span>
      <TooltipIcon>This is a TooltipIcon.</TooltipIcon>
    </Paragraph>
  </Box>
</Playground>
