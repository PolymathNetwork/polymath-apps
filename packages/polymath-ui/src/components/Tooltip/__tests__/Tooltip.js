import React from 'react';
import { render } from '../../../../testUtils';

import Tooltip from '../Tooltip';

test('renders without crashing', () => {
  const { container } = render(
    <Tooltip triggerText="Hover the icon!">This is a Tooltip.</Tooltip>
  );

  expect(container.firstChild).toMatchSnapshot();
});
