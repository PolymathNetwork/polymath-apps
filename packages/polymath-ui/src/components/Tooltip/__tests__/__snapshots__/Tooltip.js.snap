// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.c0 {
  vertical-align: middle;
  line-height: 1;
  margin-right: 4px;
}

<span
  class="c0"
>
  Hover the icon!
</span>
`;
